index.d.ts

```diff
@@ -93,8 +93,8 @@ export function combineReducers<S>(reducers: ReducersMapObject<S>): Reducer<S>;
  * transform, delay, ignore, or otherwise interpret actions or async actions
  * before passing them to the next middleware.
  */
-export interface Dispatch<S> {
-    <A extends Action>(action: A): A;
+export interface Dispatch<A extends Action = Action> {
```

Comment:
How about declare `Dispatch` like this:
```ts
export interface Dispatch<TInput = Action, TReturn = Action> {
    (input: TInput): TReturn;
}
```
for `redux-thunk` user, can use like `store.dispatch<() => Action>(() => ({ type: "action" }))`
for `redux-promise` user, can use like `store.dispatch<Action, Promise<Action>>(someAction)`